```
Parse error: ./lib/private/Session/Internal.php:160
    158| 	}
    159| 
  > 160| 	private function start(): void {
    161| 		if (@\session_id() === '') {
    162| 			// prevents javascript from accessing php session cookies
Unexpected ':', expecting ';' or '{' in ./lib/private/Session/Internal.php on line 160
Makefile:206: recipe for target 'test-php-lint' failed
make: *** [test-php-lint] Error 1
```

and phan failure

--- a/lib/base.php
+++ b/lib/base.php
@@ -411,13 +411,6 @@ class OC {
 	}
 
 	public static function initSession() {
-		// prevents javascript from accessing php session cookies
-		\ini_set('session.cookie_httponly', true);
-
-		// set the cookie path to the ownCloud directory
-		$cookie_path = OC::$WEBROOT ? : '/';
-		\ini_set('session.cookie_path', $cookie_path);
-
 		// Let the session name be changed in the initSession Hook
 		$sessionName = OC_Util::getInstanceId();
 
@@ -596,9 +589,6 @@ class OC {
 		self::checkInstalled();
 
 		OC_Response::addSecurityHeaders();
-		if (self::$server->getRequest()->getServerProtocol() === 'https') {
-			\ini_set('session.cookie_secure', true);
-		}
 
 		if (!\defined('OC_CONSOLE')) {
 			$errors = OC_Util::checkServer(\OC::$server->getConfig());

--- a/lib/private/Session/Internal.php
+++ b/lib/private/Session/Internal.php
@@ -174,6 +174,10 @@ class Internal extends Session {
 	}
 
 	private function getServerProtocol() {
+		$method = null;
+		if (isset($_SERVER['REQUEST_METHOD'])) {
+			$method = $_SERVER['REQUEST_METHOD'];
+		}
 		$req = new Request(
 			[
 				'get' => $_GET,
@@ -182,7 +186,7 @@ class Internal extends Session {
 				'server' => $_SERVER,
 				'env' => $_ENV,
 				'cookies' => $_COOKIE,
-				'method' => $_SERVER['REQUEST_METHOD'] ?? null,
+				'method' => $method,
 				'urlParams' => [],
 			],
 			null,

--- a/lib/private/Session/Internal.php
+++ b/lib/private/Session/Internal.php
@@ -27,6 +27,7 @@
 
 namespace OC\Session;
 
+use OC\AppFramework\Http\Request;
 use OCP\Session\Exceptions\SessionNotAvailableException;
 
 /**
@@ -45,12 +46,12 @@ class Internal extends Session {
 		\session_name($name);
 		\set_error_handler([$this, 'trapError']);
 		try {
-			\session_start();
+			$this->start();
 		} catch (\Exception $e) {
 			\setcookie(\session_name(), null, -1, \OC::$WEBROOT ? : '/');
 		}
 		\restore_error_handler();
-		if (!isset($_SESSION)) {
+		if ($_SESSION === null) {
 			throw new \Exception('Failed to start session');
 		}
 	}
@@ -58,6 +59,7 @@ class Internal extends Session {
 	/**
 	 * @param string $key
 	 * @param integer $value
+	 * @throws \Exception
 	 */
 	public function set($key, $value) {
 		$this->validateSession();
@@ -153,5 +155,41 @@ class Internal extends Session {
 		if ($this->sessionClosed) {
 			throw new SessionNotAvailableException('Session has been closed - no further changes to the session are allowed');
 		}
+	}
+
+	private function start() {
+		if (@\session_id() === '') {
+			// prevents javascript from accessing php session cookies
+			\ini_set('session.cookie_httponly', true);
+
+			// set the cookie path to the ownCloud directory
+			$cookie_path = \OC::$WEBROOT ? : '/';
+			\ini_set('session.cookie_path', $cookie_path);
+
+			if ($this->getServerProtocol() === 'https') {
+				\ini_set('session.cookie_secure', true);
+			}
+		}
+		\session_start();
+	}
+
+	private function getServerProtocol() {
+		$req = new Request(
+			[
+				'get' => $_GET,
+				'post' => $_POST,
+				'files' => $_FILES,
+				'server' => $_SERVER,
+				'env' => $_ENV,
+				'cookies' => $_COOKIE,
+				'method' => $_SERVER['REQUEST_METHOD'] ?? null,
+				'urlParams' => [],
+			],
+			null,
+			\OC::$server->getConfig(),
+			null
+		);
+
+		return $req->getServerProtocol();
 	}
 }

